distributed/bokeh/components.py

```diff
@@ -382,8 +386,10 @@ def cb(attr, old, new):
                 data = profile.plot_data(self.states[ind], profile_interval)
                 del self.states[:]
                 self.states.extend(data.pop('states'))
+                changing[0] = True  # don't recursively trigger callback
                 self.source.data.update(data)
                 self.source.selected = old
```

Comment:
There is still a styling issue.  I'd like to clear the selected glyphs.  See https://stackoverflow.com/questions/50819338/clear-selection-from-bokeh-plot

Reply:
Styling issue seems to have been resolved.